### ISSUE TITLE
progress error

### ISSUE THREAD

**Issue by ryan1725** [nabil6391/flutter_download_manager#12]
Problem:
The task was paused first, then resumed, repeated several times, and the initial progress was found to be wrong when resumed。Restoration progress will start from partialFile.length()

https://user-images.githubusercontent.com/119268786/221539959-e5af41bc-b3b1-41cc-9a46-1ba549c4fd86.mp4

Reason:
The reason is that the current progress is not written to the partialFile during the pause。

Sloved may be:
```
Future<void> download(String url, String savePath, cancelToken,
      {forceDownload = false}) async {
   + var partialFilePath = savePath + partialExtension;
   + late File partialFile;

      - //var partialFilePath = savePath + partialExtension;
      - //var partialFile = File(partialFilePath);
      + partialFile = File(partialFilePath);
----------------------------------------------------------------------------------------------------
catch (e) {
      var task = getDownload(url)!;
      if (task.status.value != DownloadStatus.canceled &&
          task.status.value != DownloadStatus.paused) {
        setStatus(task, DownloadStatus.failed);
        runningTasks--;

        if (_queue.isNotEmpty) {
          _startExecution();
        }
        rethrow;
     // The condition add e instance of CancelToken？if other error, seems to be ok。。。。
      } else if (task.status.value == DownloadStatus.paused) {
        + var ioSink = partialFile.openWrite(mode: FileMode.writeOnlyAppend);
        + var _f = File(partialFilePath + tempExtension);
        + await ioSink.addStream(_f.openRead());
        + await ioSink.close();

      }
    }
```

**Comment by nabil6391** [comment]
Can you create a PR?

**Comment by xuge2it** [comment]
> Can you create a PR?
[fixed initial progress error during resume, after multiple pause resumes by xuge2it · Pull Request #13 · nabil6391/flutter_download_manager](https://github.com/nabil6391/flutter_download_manager/pull/13)

**Comment by jinfagang** [comment]
Does it merge?

**Comment by nabil6391** [comment]
Released now 0.5.5
